src/qibo/models/dbi/utils.py

```diff
@@ -0,0 +1,163 @@
+from copy import deepcopy
+from itertools import product
+from typing import Optional
+
+import numpy as np
+from hyperopt import hp, tpe
+
+from qibo import symbols
+from qibo.config import raise_error
+from qibo.hamiltonians import Hamiltonian, SymbolicHamiltonian
+from qibo.models.dbi.double_bracket import (
+    DoubleBracketGeneratorType,
+    DoubleBracketIteration,
+)
+
+
+def generate_Z_operators(nqubits: int):
+    """Generate a dictionary containing 1) all possible products of Pauli Z operators for L = n_qubits and 2) their respective names.
```

Comment:
Might want to say what is the type of the values in the dict, SymbolicHamiltonian or?